Cargo.toml

```diff
@@ -122,6 +122,12 @@ either = "1.0"
 fst = { version = "0.4.7", features = ["levenshtein"] }
 fsst = { version = "=0.38.2", path = "./rust/compression/fsst" }
 futures = "0.3"
+#geoarrow-array = "0.5.0"
+#geoarrow-schema = "0.5.0"
+geoarrow-array = { git = "https://github.com/geoarrow/geoarrow-rs", rev = "61a535b072766003ad06a4d7a25dcc15f010e68f", version = "0.5" }
```

Comment:
I think we should at least make this a dedicated feature `geo` instead of directly in the main cargo. But even with that, we are inflating the dependency further because we have to have all features to do the python binding, which is against what we wanna do in #2224. 

This makes me think should we make this a separated `lance-geo` (or `geolance`?) package? That also matches the pattern of having `geoarrow`, `geoparquet`, etc. as separated projects. What do we think? @westonpace @wjones127 @Xuanwo

Reply:
Or even more simply, we can just make these documentations, so users know how to use it in python & rust, what package to import, how to construct Arrow schema, what DataFusion UDFs to register, etc. We don't really need to have any dedicated code integrations until at some point in the future we are thinking about doing any special encoding for these Geo types, but that seems unlikely for now.

Reply:
Also cc @jaystarshot curious about your opinion here as well.